app/models/payment_intent.rb

```diff
@@ -85,16 +86,16 @@ def update_status_and_charges(payment_account, api_intent, action_source, source
       end
 
       # Write any additional details now that the main status cycle is complete
-      self.update_intent_details(api_intent)
+      self.update_intent_details(updated_wca_status, api_intent)
```

Comment:
If you pass this along here, then a lot of `update` statements above are becoming partially redundant (they also execute an update on `wca_status` among others, which could be cleaned up)